### PR TITLE
deprecate `writemime` to methods of `show`

### PR DESCRIPTION
**Pull request by JeffBezanson**
This is the next bit of #14052.

The one slight wart is that text/plain showing defaults to calling `show` with 2 arguments, so you should never define text/plain showing directly, but instead just add a 2-argument method. ~~Which reminds me to update the docs.~~ done.

### CODE REVIEW

File: doc/stdlib/io-network.rst [hunk 1]
```diff
@@ -703,27 +703,29 @@ Julia environments (such as the IPython-based IJulia notebook).
...
+   The ``display`` functions ultimately call ``show`` in order to write an object ``x`` as a given ``mime`` type to a given I/O ``stream`` (usually a memory buffer), if possible. In order to provide a rich multimedia representation of a user-defined type ``T``\ , it is only necessary to define a new ``show`` method for ``T``\ , via: ``show(stream, ::MIME"mime", x::T) = ...``\ , where ``mime`` is a MIME-type string and the function body calls ``write`` (or similar) to write that representation of ``x`` to ``stream``\ . (Note that the ``MIME""`` notation only supports literal strings; to construct ``MIME`` types in a more flexible manner use ``MIME{Symbol("")}``\ .)
```
Comment (H-225) [inline doc/stdlib/io-network.rst]: > it is only necessary to define a new ``show`` method for ``T``\ , via: ``show(stream, ::MIME"mime", x::T)

Mention that users who want to overload `show` for `text/plain` should overload the 2-arg version?

Comment (H-225) [inline doc/stdlib/io-network.rst]: Wait, it's mentioned below at L716.
Mention the `text/plain` fallback, immediately after this quote instead? :smile:

File: base/docs/helpdb/Base.jl [hunk 2]
```diff
@@ -6662,28 +6662,32 @@ Return, but do not print, the time elapsed since the last [`tic`](:func:`tic`).
...
+to `import Base.show` in order to add new methods to the built-in Julia function
+`show`.
 
 Technically, the `MIME"mime"` macro defines a singleton type for the given `mime` string,
 which allows us to exploit Julia's dispatch mechanisms in determining how to display objects
 of any given type.
+
+The default MIME type is `MIME"text/plain"`. There is a fallback definition for `text/plain`
+output that calls `show` with 2 arguments. Therefore, this case should be handled by
+defining a 2-argument `show` method.
```
Comment (stevengj) [inline base/docs/helpdb/Base.jl]: A `2-argument` `show(stream::IO, x::MyType)` method.

**Comment by stevengj** [comment]
Is this breaking? If a module defines writemime, will that now be silently ignored?

**Comment by JeffBezanson** [comment]
Ah, it looks like I should have used deprecate_binding. Then it should work.

**Comment by stevengj** [comment]
Thanks! (fixed in d42585ce35b2f07ee88ceecc6076eac30bba6205)

**Comment by dancasimiro** [comment]
How do I support Julia v0.4 and v0.5 in a package? Will support be added to Compat?

**Comment by stevengj** [comment]
Yes, `@compat` will have to do it.

**Pull request merged**